internal/worker/jwtparser/worker.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/internal/jwtparser"
+)
+
+type jwtParserWorker struct {
+	tomb      tomb.Tomb
+	jwtParser *jwtparser.Parser
+}
+
+// ControllerConfig defines an interface to retrieve controller config.
+type ControllerConfig interface {
+	ControllerConfig() (controller.Config, error)
+}
+
+// HTTPClient defines an interface for an HTTP client
+// with a single GET method.
+type HTTPClient interface {
```

Comment:
Do we need to define our own interface that just includes jwk.HTTPClient? it seems to me that we gain nothing by doing this and could just use jwk type instead.. If we are going to define our own interface, then let's just put a GET method on it.

Reply:
I'd prefer to leave this as is for now, as it helps avoids importing the jwk package in another place.